src/terrain.cpp

```diff
@@ -199,7 +199,7 @@ terrain_type::terrain_type(const terrain_type& base, const terrain_type& overlay
 	editor_image_(base.editor_image_ + "~BLIT(" + overlay.editor_image_ +")"),
 	id_(base.id_+"^"+overlay.id_),
 	name_(overlay.name_),
-	editor_name_(base.editor_name_ + " / " + overlay.editor_name_),
+	editor_name_(base.editor_name_ + " / " + (overlay.editor_name_.empty() ? overlay.name_ : overlay.editor_name_)),	// Resolve Bug #23023: fall back to overlay name if overlay editor name is empty
```

Comment:
``base.editor_name_`` in this constructor has a chance to suffer from the same problem (being empty), doesn't it? Probably only in add-ons rather than mainline.